source/opt/fold.cpp

```diff
@@ -624,14 +673,18 @@ bool IsFoldableType(ir::Instruction* type_inst) {
 
 ir::Instruction* FoldInstruction(ir::Instruction* inst,
                                  std::function<uint32_t(uint32_t)> id_map) {
-  ir::Instruction* folded_inst = FoldInstructionToConstant(inst, id_map);
-  if (folded_inst != nullptr) {
-    return folded_inst;
+  InstructionBuilder<ir::IRContext::kAnalysisDefUse> ir_builder(inst->context(),
+                                                                inst);
+  bool kill_inst = false;
+  ir::Instruction* folded_inst = nullptr;
+  ir::Instruction* next_attempt =
```

Comment:
You can fold this into the condition test of the while loop.

Reply:
Yes, that does look better.